src/components/Navbar/Navbar.js

```diff
@@ -30,43 +42,72 @@ class NavBar extends React.Component {
   };
 
   render() {
+    const {logo} = this.props;
     const {showMenu} = this.state;
 
+    const screenWidth =
+      window.innerWidth ||
+      document.documentElement.clientWidth ||
+      document.body.clientWidth;
+
+    const mobileBreakPoint = 768;
+
+    const brandLogo = <div className="navbar-brand">{logo}</div>;
+
+    const itemSearchBar = (
+      <div className="navbar-search-container">
+        A search component will be inserted here.
+      </div>
+    );
+
+    const bookShelfLink = (
+      <Link to={ROUTES.MY_BOOK_HISTORY} className="navbar-bookshelves">
+        <FontAwesomeIcon icon={faBook} />
+      </Link>
+    );
+
+    const accountLink = (
+      <Link to={ROUTES.ACCOUNT} className="navbar-account">
+        <FontAwesomeIcon icon={faUser} />
+      </Link>
+    );
+
+    const mobileMenuButton = (
+      <Button
+        className="navbar-mobile-menu"
+        onClick={this.onMenuToggle}
+        text={
+          <React.Fragment>
+            <FontAwesomeIcon icon={faBars} />
+            <FontAwesomeIcon icon={faTimes} />
+          </React.Fragment>
+        }
+      />
+    );
+
     return (
       <nav className={showMenu ? 'navbar mobile-menu-open' : 'navbar'}>
         <div className="navbar-content-container">
-          <div className="navbar-brand">{this.props.logo}</div>
-          <div className="navbar-search-container">
-            A search component will be inserted here.
-          </div>
-          <Link to={MY_BOOK_HISTORY} className="navbar-bookshelves">
-            <FontAwesomeIcon icon={faBook} />
-          </Link>
-          <Link to={ACCOUNT} className="navbar-account">
-            <FontAwesomeIcon icon={faUser} />
-          </Link>
-          <Button
-            className="navbar-mobile-menu"
-            onClick={this.onMenuToggle}
-            text={
-              <React.Fragment>
-                <FontAwesomeIcon icon={faBars} />
-                <FontAwesomeIcon icon={faTimes} />
-              </React.Fragment>
-            }
-          />
+          {brandLogo}
+          {itemSearchBar}
+          {bookShelfLink}
+          {accountLink}
+          {screenWidth < mobileBreakPoint ? mobileMenuButton : null}
+          <AuthUserContext.Consumer>
+            {authUser => (authUser ? <NavbarAuth /> : <NavbarNonAuth />)}
+          </AuthUserContext.Consumer>
         </div>
       </nav>
     );
   }
 }
 
-NavBar.propTypes = {
+Navbar.propTypes = {
   logo: PropTypes.string
 };
 
-NavBar.defaultProps = {
+Navbar.defaultProps = {
```

Comment:
We could use either React Hooks or @Decorators to improve this part, setting us propTypes and defaultProps at the end of the file, which is not convenient.

Reply:
I will look into this right now.